fml/trace_event.h

```diff
@@ -70,6 +71,17 @@
   ::fml::tracing::TraceCounter((category_group), (name), (counter_id), (arg1), \
                                __VA_ARGS__);
 
+// Certain trace events happen so frequently that in complex apps the tracing
+// overhead can easily dwarf the actual raster time. For these, we use
+// a special debug mode only trace macro.
+#if (FLUTTER_RUNTIME_MODE_DEBUG)
+#define DEBUG_TRACE_EVENT0(category_group, name)     \
```

Comment:
It'd be nice if there was some way to turn these on or off via a shell argument, kind of like how we have `--trace-skia` to enable extra skia tracing today.

Having these in a trace can be helpful to understand scene complexity, but maybe there are better ways we can tool that (like @iskakaushik 's work around rasterizing leaves and timing that).

Reply:
The argument to turn this kind of thing on is `--profile`.

Reply:
Leaving these on by default in profile mode is likely artificially slowing down benchmarks, especially on iOS where tracing is more expensive.

I'm suggesting treating these more like how we treat Skia's helpful but too-verbose tracing in profile mode.

Reply:
We could even leave them on by default, but we need to provide a mechanism to turn them off so our benchmarks are representative